daemon/algod/api/server/v2/handlers.go

```diff
@@ -364,6 +319,68 @@ func (v2 *Handlers) RawTransaction(ctx echo.Context) error {
 	return ctx.JSON(http.StatusOK, generated.PostTransactionsResponse{TxId: txid.String()})
 }
 
+// TealDryRun takes transactions and additional simulated ledger state and returns debugging information.
+// (POST /v2/teal/dryrun)
+func (v2 *Handlers) TealDryRun(ctx echo.Context) error {
+	if !v2.Node.Config().EnableDeveloperAPI {
+		return ctx.String(http.StatusNotFound, "/teal/dryrun was not enabled in the configuration file by setting the EnableDeveloperAPI to true")
+	}
+	req := ctx.Request()
+	buf := new(bytes.Buffer)
+	req.Body = http.MaxBytesReader(nil, req.Body, maxTealDryrunBytes)
+	buf.ReadFrom(req.Body)
+	data := buf.Bytes()
+
+	var dr DryrunRequest
+	var gdr generated.DryrunRequest
+	err := decode(protocol.JSONHandle, data, &gdr)
+	if err == nil {
+		dr, err = DryrunRequestFromGenerated(&gdr)
+		if err != nil {
+			return badRequest(ctx, err, err.Error(), v2.Log)
+		}
+	} else {
+		err = decode(protocol.CodecHandle, data, &dr)
+		if err != nil {
+			return badRequest(ctx, err, err.Error(), v2.Log)
+		}
+	}
+
+	// fetch previous block header just once to prevent racing with network
+	var hdr bookkeeping.BlockHeader
+	if dr.ProtocolVersion == "" || dr.Round == 0 || dr.LatestTimestamp == 0 {
+		actualLedger := v2.Node.Ledger()
+		hdr, err = actualLedger.BlockHdr(actualLedger.Latest())
+		if err != nil {
+			return internalError(ctx, err, "current block error", v2.Log)
+		}
+	}
+
+	var response generated.DryrunResponse
+
+	var proto config.ConsensusParams
+	if dr.ProtocolVersion != "" {
+		var ok bool
+		proto, ok = config.Consensus[protocol.ConsensusVersion(dr.ProtocolVersion)]
+		if !ok {
+			return badRequest(ctx, nil, "invalid protocol version", v2.Log)
```

Comment:
could you change it from `"invalid protocol version"` -> `"unsupported protocol version"` ? 
The protocol could be very much valid one, but just unsupported by current binary.

also, when issuing the response, add the provided protocol version that was sent in the request.

Reply:
done